Add HT42B534 USB to UART IC

thatoddmailbox opened the pull request:
Datasheet: https://www.holtek.com/documents/10179/11842/HT42B534-xv110.pdf

There are four different packages, and they each have different pinouts, so I made four symbols. The datasheet doesn't give any different part numbers for the packages, so I just added Holtek's names for the packages to end of the symbol names.

10MSOP:
![image](https://user-images.githubusercontent.com/1904587/80925627-638fcf00-8d5f-11ea-8b0b-d9938c68bea6.png)

10SOP:
![image](https://user-images.githubusercontent.com/1904587/80925634-6d193700-8d5f-11ea-9731-969b4852b927.png)

16NSOP:
![image](https://user-images.githubusercontent.com/1904587/80925639-76a29f00-8d5f-11ea-91de-c33e5e169cc0.png)

8SOP:
![image](https://user-images.githubusercontent.com/1904587/80925643-802c0700-8d5f-11ea-9445-74307bae61df.png)


---

All contributions to the kicad library must follow the [KiCad library convention](http://kicad-pcb.org/libraries/klc/)

Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [X] Provide a URL to a datasheet for the symbol(s) you are contributing
- [X] An example screenshot image is very helpful
- [X] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
  - A new fitting footprint must be submitted if the library does not yet contain one.
- [X] ~If there are matching footprint PRs, provide link(s) as appropriate~
- [ ] Check the output of the Travis automated check scripts - fix any errors as required
- [ ] Give a reason behind any intentional library convention rule violation.

myfreescalewebpage commented:
Hello @thatoddmailbox and thanks for these symbols

4 symbols is the correct solution. Can you please fix the branch conflict to permit starting the review ?

Thanks
Joel

thatoddmailbox commented:
@myfreescalewebpage OK, the conflict should be fixed now!

myfreescalewebpage commented:
Hi @thatoddmailbox , thanks!

A few comments I have during my review:

HT42B534-x-8SOP
- [ ] Description should ends with package name: `USB to UART Bridge IC, SOP-8`

HT42B534-x-10MSOP
- [ ] Description should ends with package name: `USB to UART Bridge IC, MSOP-10`

HT42B534-x-10SOP
- [ ] Description should ends with package name: `USB to UART Bridge IC, SOP-10`
- [ ] Footprint should be SOP: `Package_SO:SOP-10_3.9x4.9mm_P1.00mm` and fp filter `SOP*3.9x4.9mm*P1.00mm*`

HT42B534-x-16NSOP
- [ ] Description should ends with package name: `USB to UART Bridge IC, NSOP-16`
- [ ] Footprint should be SOP: `Package_SO:NSOP-16_3.9x9.9mm_P1.27mm` and fp filter `NSOP*3.9x9.9mm*P1.27mm*`

Important note: repo will be locked the 2020-09-01. No new PR will be possible until KiCad V6 release. If you want to have the symbols merged quickly, you should propose the missing footprints ASAP.

Cheers,
Joel

thatoddmailbox commented:
@myfreescalewebpage OK, I've updated the descriptions and changed the footprint names. What I ended up doing was copying the existing footprints and renaming them to match Holtek's name. You can compare the Holtek datasheet with the datasheets for the original footprints and see that the dimensions are the same.

The pull requests for the symbols is: KiCad/kicad-footprints#2432

myfreescalewebpage commented:
Thanks, no more comment here, just need to wait the footprints.

cpresser commented:
I am not sure that I agree that we should put duplicates of footprints into the footprint-repo just because some manufacturer choose to rename them. That just generates more uncertainty for the user, maintaining effort for us and is just bloat.
How about putting both package names into the description of the symbol and adding using the standard footprint and filter.

myfreescalewebpage commented:
@cpresser I understand the position.
But the issue is : is it 100% identical ? even the height of the device for the package 3D ?
Moreover, later when users are taking the part in the library, if the footprint name do not correspond, they may have some doubts about the definition of the symbol.
The same issue exists with SO, SOP and SOIC for example.
We have no clear rules about this problem at the moment.
Joel

myfreescalewebpage commented:
No news of the author on the fp, indicate this is Abandoned
